### PR TITLE
New updating for preview php content

### PR DESCRIPTION
**Pull request by lamrani002**
# NEW|New updating for preview php content on Ckeditor 
for exemple preview news : 
gogog

### CODE REVIEW

File: htdocs/core/lib/process_template.lib.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<?php
```
Comment (eldy) [inline htdocs/core/lib/process_template.lib.php:1]: This is the ajax service i think ?
If i am right, move the file into directory htdocs/core/ajax/emailingemplates.php.

File: htdocs/core/lib/process_template.lib.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+if (!defined('NOREQUIRESOC')) {
+	define('NOREQUIRESOC', '1');
+}
+require_once '../../main.inc.php';
+require_once './files.lib.php';
+
+top_httphead();
+
+if ($_SERVER['REQUEST_METHOD'] == 'POST' && GETPOSTISSET('content')) {
+	$content = $_POST['content'];
```
Comment (eldy) [inline htdocs/core/lib/process_template.lib.php:27]: $_POST use is forbidden.
You must use GETPOST()

File: htdocs/core/lib/process_template.lib.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	$content = str_replace('<!-- PHP_END -->', ' ?>', $content);
+
+	$directory = DOL_DATA_ROOT . '/mailing/email_template';
+	if (!is_dir($directory)) {
+		dol_mkdir($directory);
+	}
+
+	$i = 0;
+	do {
+		$filePath = $directory . '/template_' . $i++ . '.php';
```
Comment (eldy) [inline htdocs/core/lib/process_template.lib.php:40]: The $directory = DOL_DATA_ROOT . '/mailing/email_template'; 
is inside the root where binary files are stored. It means the directory should be in readonly mode and is not writable by the web server on a secured installation.
Any temporaty files must be stored into a temp directory.
For example, you can use  $conf->admin->dir_temp

Comment (lamrani002) [reply htdocs/core/lib/process_template.lib.php]: ok it s solved

**Pull request closed without merging**